platforms/access-bridge/src/services/access-service.ts

```diff
@@ -0,0 +1,118 @@
+  RefreshAccessTokensParams,
+} from '@jwp/ott-common/types/access.js';
+
+import { ACCESS_CONTROL_CLIENT, API_SECRET } from '../app-config.js';
+import { BadRequestError, ForbiddenError, NotFoundError, ParameterInvalidError, isJWError } from '../errors.js';
+import { put } from '../http.js';
+
+/**
+ * AccessService handles interactions with the access management APIs.
+ * It provides methods to generate and refresh access tokens.
+ */
+export class AccessService {
+  private apiSecret: string;
+  private accessControlClient: string;
+
+  constructor() {
+    this.apiSecret = API_SECRET;
+    this.accessControlClient = ACCESS_CONTROL_CLIENT;
+  }
+
+  /**
+   * Generate access tokens for a specific site and subscriber.
+   * @param siteId The ID of the site.
+   * @param email The subscriber's email address.
+   * @param plans Array of access plans for the subscriber.
+   * @returns A Promise resolving to an AccessTokensResponse.
+   * @throws ForbiddenError if the request is not properly authenticated.
+   * @throws BadRequestError for other validation error scenarios.
+   */
+  async generateAccessTokens({ siteId, viewerId, plans }: GenerateAccessTokensParams): Promise<AccessTokensResponse> {
+    try {
+      const url = await this.generateSignedUrl(`/v2/sites/${siteId}/access/generate`);
+      const payload = {
+        subscriber_info: {
+          email: viewerId,
+          plans,
+        },
+      };
+
+      return await put<AccessTokensResponse, typeof payload>(url, payload);
+    } catch (e) {
+      if (isJWError(e)) {
+        const error = e.errors[0];
+        // Possible error scenarios coming from Delivery Gateway
+        switch (error.code) {
+          case 'forbidden':
+            throw new ForbiddenError({ description: error.description });
+          default:
+            throw new BadRequestError({ description: error.description });
+        }
+      }
+      console.error('Service: error generating access tokens:', e);
+      throw e;
+    }
+  }
+
+  /**
+   * Refresh access tokens for a specific site using a refresh token.
+   * @param siteId The ID of the site.
+   * @param refreshToken The refresh token to use for token refresh.
+   * @returns A Promise resolving to an AccessTokensResponse.
+   * @throws ForbiddenError if the request is not properly authenticated.
+   * @throws ParameterInvalidError if the refresh token is missing or invalid.
+   * @throws NotFoundError if the requested resource is not found.
+   * @throws BadRequestError for other validation error scenarios.
+   */
+  async refreshAccessTokens({ siteId, refreshToken }: RefreshAccessTokensParams): Promise<AccessTokensResponse> {
+    try {
+      const url = await this.generateSignedUrl(`/v2/sites/${siteId}/access/refresh`);
+      const payload = {
+        refresh_token: refreshToken,
+      };
+
+      return await put<AccessTokensResponse, typeof payload>(url, payload);
+    } catch (e) {
+      if (isJWError(e)) {
```

Comment:
Is there a reason to define these switch cases for each method? This seems like something that can be universal.

Reply:
Good catch, defined a global handler for this that uses the entire error definition object.